db/migrate/20150226095809_add_status_enums_to_orphan.rb

```diff
@@ -0,0 +1,30 @@
+class AddStatusEnumsToOrphan < ActiveRecord::Migration
+  def up
+    add_column :orphans, :status, :integer, default: 0
+    add_column :orphans, :sponsorship_status, :integer, default: 0
+
+    set_status
+    set_sponsorship_status
+  end
+
+  def down
+    remove_column :orphans, :status
+    remove_column :orphans, :sponsorship_status
+  end
+
+  private
+
+  def set_status
+    Orphan.all.each do |orphan|
+      status_code = orphan.orphan_status.code
+      orphan.update_column(:status, ( status_code - 1 ))
+    end
+  end
+
+  def set_sponsorship_status
+    Orphan.all.each do |orphan|
+      status_code = orphan.orphan_sponsorship_status.code
+      orphan.update_column(:sponsorship_status, ( status_code - 1 ))
+    end
+  end
+end
```

Comment:
This migration has to do the important task of preserving data integrity and correctly populating the new enum statuses based on previous associations. I had written a spec for it, but since the test was tied to migration history, it was going to be very brittle. Indeed, after rebasing on latest develop, the spec not only failed but also broke the remainder of the test suite, and I had to take it out. You can see the code [here](https://gist.github.com/6dc9c81bab00c484ca82.git).